contrib/sly-mrepl.el

```diff
@@ -456,6 +456,27 @@ In that case, moving a sexp backward does nothing."
                          (overlay-end sly-mrepl--last-prompt-overlay)
                          '(font-lock-face sly-mrepl-prompt-face))))
 
+(defun sly-mrepl-format-default-prompt (package error-level)
+  "Return a the default SLY prompt string.
+Suitable for `sly-mrepl-prompt-formatter'."
+  (concat
+   (when (cl-plusp error-level)
+     (concat (sly-make-action-button
+              (format "[%d]" error-level)
+              #'sly-db-pop-to-debugger-maybe)
+             " "))
+   (propertize
+    (concat package "> ")
+    'face 'sly-mrepl-prompt-face
+    'font-lock-face 'sly-mrepl-prompt-face)))
+
+(defcustom sly-mrepl-prompt-formatter #'sly-mrepl--format-default-prompt
+  "Function called when preparing the prompt.
+It takes the current Common Lisp package and the error-level (possibly nil).
```

Comment:
The types for these should be specified.  I'd write "Takes two arguments, a string designating the current Common Lisp package and a fixnum indicating how many errors are outstanding.

That said, I wonder if it could take some extra information, such as the type of the last error, and the number of the value for backreference-writing purposes, etc.